python/pyspark/pandas/supported_api_gen.py

```diff
@@ -0,0 +1,340 @@
+    ps_module_group: Any,
+) -> None:
+    """
+    Updates supported status across multiple module paths.
+
+    Parmeters
+    ---------
+    all_supported_status: Dict[Tuple[str, str], Dict[str, SupportedStatus]]
+        Data that stores the supported status across multiple module paths.
+    pd_modles: List[str]
+        Name list of pandas modules.
+    pd_module_group : Any
+        Specific path of importable pandas module.
+    ps_module_group: Any
+        Specific path of importable pyspark.pandas module.
+    """
+    pd_modules += [""]  # for General Function APIs
+    for module_name in pd_modules:
+        supported_status = create_supported_by_module(module_name, pd_module_group, ps_module_group)
+        if supported_status:
+            all_supported_status[(module_name, ps_module_group.__name__)] = supported_status
+
+
+def write_table(
+    module_name: str,
+    module_path: str,
+    supported_status: Dict[str, SupportedStatus],
+    w_fd: TextIO,
+) -> None:
+    """
+    Write table by using Sphinx list-table directive.
+    """
+    lines = []
+    lines.append("Supported ")
+    if module_name:
+        lines.append(module_name)
+    else:
+        lines.append("General Function")
+    lines.append(" APIs\n")
+    lines.append("-" * 100)
+    lines.append("\n")
+    lines.append(f".. currentmodule:: {module_path}")
+    if module_name:
+        lines.append(f".{module_name}\n")
+    else:
+        lines.append("\n")
+    lines.append("\n")
+    lines.append(".. list-table::\n")
+    lines.append("    :header-rows: 1\n")
+    lines.append("\n")
+    lines.append("    * - API\n")
+    lines.append("      - Implemented\n")
+    lines.append("      - Missing parameters\n")
+    for func_str, status in supported_status.items():
+        func_str = escape_func_str(func_str)
+        if status.implemented == Implemented.NOT_IMPLEMENTED.value:
+            lines.append(f"    * - {func_str}\n")
+        else:
+            lines.append(f"    * - :func:`{func_str}`\n")
+        lines.append(f"      - {status.implemented}\n")
+        lines.append("      - \n") if not status.missing else lines.append(
+            f"      - {status.missing}\n"
+        )
+    w_fd.writelines(lines)
+
+
+def escape_func_str(func_str: str) -> str:
+    """
+    Transforms which affecting rst data format.
+    """
+    if func_str.endswith("_"):
+        return func_str[:-1] + "\_"  # noqa: W605
+    else:
+        return func_str
+
+
+def write_rst(all_supported_status: Dict[Tuple[str, str], Dict[str, SupportedStatus]]) -> None:
+    """
+    Writes the documentation to the target file path.
+    """
+    with open(TARGET_RST_FILE, "w") as w_fd:
+        w_fd.write(RST_HEADER)
+        for module_info, supported_status in all_supported_status.items():
+            module, module_path = module_info
+            if supported_status:
+                write_table(module, module_path, supported_status, w_fd)
+                w_fd.write("\n")
```

Comment:
it would need to add something like this on the bottom:

```python
def _test():
    import doctest
    import pyspark.pandas.supported_api_gen
    globs = pyspark.pandas.supported_api_gen.__dict__.copy()
    (failure_count, test_count) = doctest.testmod(pyspark.pandas.supported_api_gen, globs=globs)
    if failure_count:
        sys.exit(-1)


if __name__ == "__main__":
    _test()
```

to actually run the doctests in the CI. This is required because Apache Spark has its own testing way to run each modules in parallel.